### DISCUSSION TITLE
[A-L] (24 aprile 2024) Esercizio 2

### DISCUSSION THREAD

**Discussion by IonutCicio** [sapienzastudentsnetwork/progettazione-di-algoritmi#58] [Esercitazioni A-L]
![image](https://github.com/sapienzastudentsnetwork/progettazione-di-algoritmi/assets/40034429/4f1aba0d-8da2-4dc8-90b9-b62034ddfbd6)

**Comment by Elia-Belli** [comment]
## Update MST
Chiamiamo $E_u$ l'insieme degli archi nella lista di adiacenza di $u$, affinchè $u$ faccia parte del nuovo MST andrà aggiunto almeno un arco $e\in E_u$, più precisamente il minimo.
A questo punto dobbiamo valutare se i restanti archi in $E_u$ possano sostituire archi in $T$ in modo da ridurne il peso: il ragionamento è analogo a #57 (considerando uno alla volta gli archi $e'\in E_u$ come se fossero l'arco di peso aggiornato).
```python
def new_node_in_MST(T,E_u,u):
    m = min(E_u)
    T.add(m) #aggiungo a T il minimo tra gli archi incidenti in u

    for e in (E_u - {m}):
        update_MST(T,e) #aggiunge e se migliora il peso di T, 
                        #ed elimina l'arco più pesante nel ciclo generato dal suo inserimento
```

**Comment by daromacarlo** [comment]
![immagine](https://github.com/user-attachments/assets/3187f93f-6119-4c11-b15f-c6aba5984ef6)
